crates/accelerate/src/uc_gate.rs

```diff
@@ -160,3 +205,79 @@ pub fn uc_gate(m: &Bound<PyModule>) -> PyResult<()> {
     m.add_wrapped(wrap_pyfunction!(dec_ucg_help))?;
     Ok(())
 }
+
+#[cfg(test)]
+mod test {
+    use super::compute_2x2_eig;
+    use approx::abs_diff_eq;
+    use nalgebra::Matrix2;
+    use num_complex::{Complex64, ComplexFloat};
+    use rand::prelude::*;
+    use rand_distr::StandardNormal;
+    use rand_pcg::Pcg64Mcg;
+
+    #[inline(always)]
+    fn random_complex(rng: &mut Pcg64Mcg) -> Complex64 {
+        Complex64::new(rng.sample(StandardNormal), rng.sample(StandardNormal))
+            * std::f64::consts::FRAC_1_SQRT_2
+    }
+
+    fn random_unitaries(seed: u64, size: usize) -> impl Iterator<Item = Matrix2<Complex64>> {
+        let mut rng = Pcg64Mcg::seed_from_u64(seed);
+
+        (0..size).map(move |_| {
+            let mat: Matrix2<Complex64> = [
+                [random_complex(&mut rng), random_complex(&mut rng)],
+                [random_complex(&mut rng), random_complex(&mut rng)],
+            ]
+            .into();
+            let (q, r) = mat.qr().unpack();
```

Comment:
not sure that qr decomposition is needed. r is not necessarily diagonal (it's triangular), and can have zeroes on the diagonal.

Reply:
if you remove these lines for the qr decomposition, you should get a random matrix which is enough for the test.

Reply:
Simplified in: https://github.com/Qiskit/qiskit/pull/13956/commits/7d7b43866e255ac27c20324d80bf7bc0305cbc7d